BAT/BCAux.h

```diff
@@ -130,7 +130,11 @@ class BCTrash
 public:
     BCTrash() {}
     BCTrash(const BCTrash<T>&) {}
-    BCTrash<T>& operator=(const BCTrash<T>&) {}
+    BCTrash<T>& operator=(const BCTrash<T>&)
+    {
+        return *this;
+    }
+
```

Comment:
Uh, why define a copy operator that does nothing? If BCTrash isn't copyable, I'd prefer not to define `operator=`, or at least make it throw an exception.